### PR TITLE
[architecture] Isolate retry backoff policy from wall-clock time sources

### LINKED ISSUE #731: [architecture] Isolate retry backoff policy from wall-clock time sources
## Why / Context
Retry logic currently mixes elapsed-time policy decisions with direct clock reads. Separating policy from clock adapters improves determinism and keeps retry behavior testable.

## Scope
**In scope**
- Define a pure backoff policy evaluator independent of wall-clock calls.
- Wire one execution retry path and one monitoring timeout path to that evaluator.
- Add fake-time tests for retry boundaries and cap behavior.

**Out of scope**
- Changing business retry limits or strategy thresholds.
- Broad refactors across unrelated scheduling components.

Constraints:
- Keep changes small/mergeable.
- Deterministic tests only.
- Avoid touching unrelated generated artifacts unless required.

## Acceptance Criteria (required)
- [ ] Backoff decisions are computed by a pure policy function.
- [ ] Clock reads are confined to adapter boundaries at call sites.
- [ ] Unit tests cover deterministic retry/timeout behavior with fake time.
- [ ] Default runtime behavior remains unchanged when using real time sources.

## Implementation Notes / Pointers
**Likely files / modules:**
- `src/gpt_trader/utilities/time_provider.py`
- `src/gpt_trader/monitoring/health_checks.py`
- `src/gpt_trader/features/live_trade/execution/order_submission.py`

**Related tests:**
- `tests/unit/gpt_trader/utilities/test_time_provider.py`
- `tests/unit/gpt_trader/monitoring/test_health_checks.py`
- `tests/unit/gpt_trader/features/live_trade/execution/test_order_submission_retry.py`

**Edge cases to handle:**
- Negative or zero elapsed durations are handled safely.
- Backoff caps apply consistently after repeated retries.
- Parallel tests do not leak shared time state.

## Commands (local)
- `make fmt`
- `make lint`
- `uv run pytest -q tests/unit/gpt_trader/utilities/test_time_provider.py tests/unit/gpt_trader/monitoring/test_health_checks.py`

## PR Requirements
- PR title should match the issue.
- PR body must include: `Fixes #<issue-number>`
- CI must be green; if Agent Artifacts Freshness fails, run `uv run agent-regenerate`, then commit `var/agents/...` updates.

## Codex-Ready Checklist (for the issue creator)
- [x] Clear acceptance criteria
- [x] At least one file pointer
- [x] Commands included
- [x] No ambiguous "do the right thing" language

### PR DESCRIPTION
**Pull request by Solders-Girdles**
Fixes #731

(Automated PR created by OpenClaw PR finisher)
Branch: codex/731-architecture-isolate-retry-backoff-polic
Commit: 0fbe23ebddd694eb332991c22ec621ab2f8428a2

### CODE REVIEW

**Comment by Solders-Girdles** [comment]
@codex review

File: src/gpt_trader/monitoring/health_checks.py [hunk 1]
```diff
@@ -1040,6 +1041,37 @@ def _ratio_or_zero(numerator: int | float, denominator: int | float) -> float:
...
+    elif age_seconds < crit_delay:
+        attempt = 2
+    else:
+        attempt = 3
+
+    decision = evaluate_backoff_delay(
+        attempt=attempt,
+        base_delay=warn_delay,
+        max_delay=crit_delay,
+        multiplier=multiplier,
```
[the remark this answers is not in the document]
Comment (Solders-Girdles) [reply src/gpt_trader/monitoring/health_checks.py]: Fixed in commit 9b5dc5f1: zero-warn configs now keep immediate WARN metadata but emit CRIT metadata using the configured critical threshold (`timeout_delay_seconds=crit`, `timeout_capped=true`). Added `test_warn_zero_preserves_crit_timeout_metadata` and re-ran `uv run pytest tests/unit/gpt_trader/monitoring/test_health_checks_market_data_staleness.py tests/unit/gpt_trader/monitoring/test_health_checks_market_data_feed_staleness.py -q` (15 passed).
